### PR TITLE
feat: CLI command to get validator address by S3 bucket or KMS key ID

### LINKED ISSUE #3341: CLI should help with setting up validators
## Problem

When running validators on AWS, its hard to know what the validator address is. Also, announcing requires sending funds to the validator (signer address).

## Solution

There should be a CLI command which helps folks to get the validator address from the announcement by passing in the bucket. And then also there should be a related command to announce the validator directly.

### PR DESCRIPTION
**Pull request by aalaoui12**
### Description

This pull request creates the `hyperlane validator address` command, making it easy for users to get the address of their validator key. 

They can do this in 2 ways:
1. By specifying the validator announcement S3 bucket using the `--bucket` option
2. If the validator key is created in AWS KMS, they can get the address using the `--key-id` command.

This PR integrates [this script](https://github.com/tkporter/get-aws-kms-address/) in order to retrieve the validator address from KMS, removing the need for users to run the script manually.

In order to use this command, users need to export three environment variables: `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_REGION`, which correspond to the user's validator IAM credentials and region.

In the future, other `hyperlane validator` commands can be created in order to further simplify configuring validators.

### Drive-by changes

None other than adding the command files, and aws-sdk dependencies for S3 and KMS.

### Related issues

Fixes #3341

### Backward compatibility

Yes

### Testing

Manual testing under all conditions/edge cases

### CODE REVIEW

**Comment by avious00** [comment]
@aalaoui12 this is great! will get you a review from an engineer

**Comment by aalaoui12** [comment]
@avious00 Sounds good! My discord username is 2ofa.

File: typescript/cli/src/validator/address.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  keyId: string;
+}) {
+  if (!bucket && !keyId) {
+    throw new Error('Must provide either an S3 bucket or a KMS Key ID.');
+  }
+
+  // User must have below env variables configured beforehand
+  const accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+  const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+  const region = process.env.AWS_REGION;
```
Comment (jmrossy) [inline typescript/cli/src/validator/address.ts:26]: Please handle these required args like we handle keys here:
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/cli/src/commands/options.ts#L48

Meaning you should:
1. Allow users to set them via command args
2. Add these fields to `envScheme` 
3. Default to the env var when the arg is not set
4. Prompt the user for any required values that weren't provided

File: typescript/cli/src/validator/address.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  // https://tools.ietf.org/html/rfc5480#section-2
+  const res = EcdsaPubKey.decode(publicKey, 'der');
+  let pubKeyBuffer: Buffer = res.pubKey.data;
+
+  // The public key starts with a 0x04 prefix that needs to be removed
+  // more info: https://www.oreilly.com/library/view/mastering-ethereum/9781491971932/ch04.html
+  pubKeyBuffer = pubKeyBuffer.slice(1, pubKeyBuffer.length);
+
+  const address = ethers.utils.keccak256(pubKeyBuffer); // keccak256 hash of publicKey
+  const EthAddr = `0x${address.slice(-40)}`; // take last 20 bytes as ethereum adress
```
Comment (jmrossy) [inline typescript/cli/src/validator/address.ts:137]: Nit: just return this value directly

**Comment by aalaoui12** [comment]
> Good progress here. Please see my comments below and address the failing CI tests

Just fixed this, now users can pass the AWS parameters in using `--access-key`, `--secret-key`, and `--region` arguments. These will default to the respective .env variables if the args are not set, and they will be prompted for these values if neither exist.

Let me know if there are any other changes!

File: typescript/cli/src/commands/options.ts [hunk 3]
```diff
@@ -146,3 +146,35 @@ export const addressCommandOption = (
   description,
   demandOption,
 });
+
+/* Validator options */
+export const awsAccessKeyOption: Options = {
```
Comment (jmrossy) [inline typescript/cli/src/commands/options.ts:151]: Please rename this from awsAccessKeyOption -> awsAccessKeyCommandOption

File: typescript/cli/src/commands/options.ts [hunk 4]
```diff
@@ -146,3 +146,35 @@ export const addressCommandOption = (
...
+  type: 'string',
+  description: 'AWS access key of IAM user associated with validator',
+  default: ENV.AWS_ACCESS_KEY_ID,
+  defaultDescription: 'process.env.AWS_ACCESS_KEY_ID',
+};
+
+export const awsSecretKeyOption: Options = {
```
Comment (jmrossy) [inline typescript/cli/src/commands/options.ts:158]: awsSecretKeyCommandOption

File: typescript/cli/src/commands/options.ts [hunk 5]
```diff
@@ -146,3 +146,35 @@ export const addressCommandOption = (
...
+  type: 'string',
+  description: 'AWS secret access key of IAM user associated with validator',
+  default: ENV.AWS_SECRET_ACCESS_KEY,
+  defaultDescription: 'process.env.AWS_SECRET_ACCESS_KEY',
+};
+
+export const awsRegionOption: Options = {
```
Comment (jmrossy) [inline typescript/cli/src/commands/options.ts:165]: awsRegionCommandOption

File: typescript/cli/src/commands/options.ts [hunk 6]
```diff
@@ -146,3 +146,35 @@ export const addressCommandOption = (
...
+  type: 'string',
+  describe: 'AWS region associated with validator',
+  default: ENV.AWS_REGION,
+  defaultDescription: 'process.env.AWS_REGION',
+};
+
+export const bucketCommandOption: Options = {
```
Comment (jmrossy) [inline typescript/cli/src/commands/options.ts:172]: awsBucketCommandOption

File: typescript/cli/src/commands/options.ts [hunk 7]
```diff
@@ -146,3 +146,35 @@ export const addressCommandOption = (
...
+  type: 'string',
+  describe: 'AWS S3 bucket containing validator signatures and announcement',
+};
+
+export const keyIdCommandOption: Options = {
```
Comment (jmrossy) [inline typescript/cli/src/commands/options.ts:177]: awsKeyIdCommandOption

File: typescript/cli/src/utils/aws.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import { input } from '@inquirer/prompts';
+
+export async function getAccessKeyId(skipConfirmation: boolean) {
```
Comment (jmrossy) [inline typescript/cli/src/utils/aws.ts:3]: Please move all this code to typescript/cli/src/validator/address.ts at the bottom

File: typescript/cli/src/validator/address.ts [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+    throw new Error('No access key ID set.');
+  }
+
+  if (!secretKey) {
+    throw new Error('No secret access key set.');
+  }
+
+  if (!region) {
+    throw new Error('No AWS region set.');
+  }
```
Comment (jmrossy) [inline typescript/cli/src/validator/address.ts:45]: Please import the `assert` helper function from the `@hyperlane-xyz/utils` package and shorten these checks to:
`assert(accessKey, 'No access key ID set.')`

**Review by jmrossy — COMMENTED** [review]
Almost done! A few cosmetic things to fix and also please add a changeset describing this addition. run `yarn changeset` from the monorepo root, it will walk you through the steps. Use a `minor` change

**Comment by aalaoui12** [comment]
Fixed changes

**Comment by avious00** [comment]
hey @aalaoui12 reaching out over discord for the payout!

**Pull request merged**